core_lib/data_transform/result_to_dict.py

```diff
@@ -100,6 +106,10 @@ def result_to_dict(return_val, properties_as_dict: bool = True, callback: Callab
     else:
         results = return_val
 
+    if isinstance(return_val, mongomock.collection.Cursor):
+        results = __pymongo_to_dict(return_val)

```

Comment:
Also another comment I have we are checking for Cursor. which is a collection
When we return a single result., `find_one` dose `result_to_dict` support that?
how do you know if its a single element or and array?

Reply:
For the find it will return us cursor object but for find_one mongo will return us dict only so if the user do result_to_dict(find_one_data) it will not go inside pymongo_to_dict it will go inside dict_to_dict which is already tackled Thanks